app/controllers/topics_controller.rb

```diff
@@ -343,6 +343,7 @@ def destroy
     PostDestroyer.new(current_user, first_post, { context: params[:context] }).destroy
 
     DiscourseEvent.trigger(:topic_destroyed, topic, current_user)
+    WebHook.enqueue_hooks(:topic, topic_id: topic.id, user_id: current_user.id, category_id: params[:category_id]&.to_i)
```

Comment:
I will use the DiscourseEvent hooks instead for this instead of adding more code into the controller